src/app/qgsmaptoollabel.cpp

```diff
@@ -707,3 +792,19 @@ QgsMapToolLabel::LabelDetails::LabelDetails( const QgsLabelPosition &p )
     settings = QgsPalLayerSettings();
   }
 }
+
+bool QgsMapToolLabel::isAuxiliaryField( QgsVectorLayer *layer, int index ) const
```

Comment:
I think this should be moved to QgsVectorLayer itself - it could be handy elsewhere too

Reply:
Good point! Done.